dhcpv4/server4/server.go

```diff
@@ -133,7 +133,7 @@ func NewServer(addr *net.UDPAddr, handler Handler, opt ...ServerOpt) (*Server, e
 	}
 	if s.conn == nil {
 		var err error
-		conn, err := net.ListenUDP("udp4", addr)
+		conn, err := dhcpv4.BindConnectionToInterface(addr.String(), iface)
```

Comment:
Same as above, I think here you can use `NewIPv4UDPConn` which is already in the lib.